mindsdb/utilities/telemetry.py

```diff
@@ -0,0 +1,178 @@
+    from pathlib import Path
+
+    try:
+        # region collect datasources/integrations
+        integration_records = db.session.query(db.Integration).filter_by(company_id=None).all()
+        mindsdb_path = Path(importlib.util.find_spec("mindsdb").origin).parent
+        handlers_path = mindsdb_path.joinpath("integrations/handlers")
+
+        data_handlers = []
+        for handler_dir in handlers_path.iterdir():
+            if handler_dir.is_dir() is False or handler_dir.name.startswith("__"):
+                continue
+            init_file = handler_dir / "__init__.py"
+            if not init_file.exists():
+                continue
+            code = ast.parse(init_file.read_text())
+
+            is_data_handler = None
+            handler_name = None
+            for item in code.body:
+                if isinstance(item, ast.Assign) and isinstance(item.targets[0], ast.Name):
+                    name = item.targets[0].id
+                    if isinstance(item.value, ast.Attribute) and name == "type":
+                        is_data_handler = item.value.attr == "DATA"
+                        if handler_name is not None:
+                            break
+                    if isinstance(item.value, ast.Constant) and name == "name":
+                        handler_name = item.value.value
+                        if is_data_handler is not None:
+                            break
+            if is_data_handler:
+                data_handlers.append(handler_name)
+        integration_engines = [record.engine for record in integration_records if record.engine in data_handlers]
+        integrations_dict = {item: integration_engines.count(item) for item in set(integration_engines)}
+        # endregion
+
+        # region collect knowledge bases
+        knowledge_base_records = db.session.query(db.KnowledgeBase).all()
+        kb_result = []
+        for kb in knowledge_base_records:
+            params = kb.params or {}
+            kb_result.append(
+                {
+                    "embedding_model": {
+                        k: v for k, v in params.get("embedding_model", {}).items() if k in ("provider", "model_name")
+                    },
+                    "reranking_model": {
+                        k: v for k, v in params.get("reranking_model", {}).items() if k in ("provider", "model_name")
+                    },
+                }
+            )
+        # endregion
+
+        # region collect agents
+        agent_records = db.session.query(db.Agents).filter_by(company_id=None).all()
+        agents_list = []
+        for agent in agent_records:
+            params = agent.params or {}
+            model_params = params.get("model", {})
+            agents_list.append({k: v for k, v in model_params.items() if k in ("provider", "model_name")})
+        # endregion
+    except Exception as e:
+        db.session.rollback()
+        logger.error(f"Error collecting telemetry data for 'start' event: {e}")
```

Comment:
if exception happens, some variables below might not be defined for example 'agents_list'